x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/create_source_editor.js

```diff
@@ -58,41 +66,28 @@ export class CreateSourceEditor extends Component {
 
   componentDidMount() {
     this._isMounted = true;
-    this.loadIndexPattern(this.state.indexPatternId);
   }
 
-  onIndexPatternSelect = indexPatternId => {
+  _onIndexPatternSelect = indexPatternId => {
     this.setState(
       {
         indexPatternId,
       },
-      this.loadIndexPattern(indexPatternId)
+      this._loadIndexPattern(indexPatternId)
     );
   };
 
-  loadIndexPattern = indexPatternId => {
+  _loadIndexPattern = indexPatternId => {
     this.setState(
       {
         isLoadingIndexPattern: true,
         ...RESET_INDEX_PATTERN_STATE,
       },
-      this.debouncedLoad.bind(null, indexPatternId)
+      this._debouncedLoad.bind(null, indexPatternId)
     );
   };
 
-  loadIndexDocCount = async indexPatternTitle => {
```

Comment:
I think we can remove the corresponding route: https://github.com/elastic/kibana/blob/928454afa4ce0f7136c90b2a8756e0600b55e360/x-pack/legacy/plugins/maps/server/routes.js#L412-L430